md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java

```diff
@@ -179,6 +181,67 @@ public void delete(GeneralConfigId generalConfigId) {
+                    if (jpaGeneralConfig.getRequired())
+                        generalConfig.setRequired(1);
+                    else
+                        generalConfig.setRequired(0);
+                    generalConfig.setConfigGroup(jpaGeneralConfig.getId().getConfigGroup());
+                    generalConfig.setKey(jpaGeneralConfig.getId().getGcKey());
+                    generalConfig.setValue(jpaGeneralConfig.getGcValue());
+                    generalConfig.setDescription(jpaGeneralConfig.getDescription());
+                    generalConfig.setDefaultVal(jpaGeneralConfig.getDefaultVal());
+                    generalConfig.setEnabled(jpaGeneralConfig.getEnabled());
+                    generalConfig.setType(jpaGeneralConfig.getType());
+
+                    generalConfigList.add(generalConfig);
+                }
+            } else {
+                // only difference in both criteria is that requiredRestriction is added in this one.
+                boolean req = (required == 1) ? true : false;
+                Criteria getGeneralConfigCriteria = session.createCriteria(GeneralConfig.class).add(Restrictions.like(CONFIG_GROUP, "cluster%")).add(Restrictions.eq("description", description)).add(Restrictions.eq("enabled", true)).add(Restrictions.eq("required", req));
+                List<GeneralConfig> jpaGeneralConfigList = getGeneralConfigCriteria.list();
+                for (GeneralConfig jpaGeneralConfig : jpaGeneralConfigList) {
+                    com.wipro.ats.bdre.md.beans.table.GeneralConfig generalConfig = new com.wipro.ats.bdre.md.beans.table.GeneralConfig();
+                    generalConfig.setCounter(jpaGeneralConfigList.size());
+                    if (jpaGeneralConfig.getRequired())
+                        generalConfig.setRequired(1);
+                    else
+                        generalConfig.setRequired(0);
+                    generalConfig.setConfigGroup(jpaGeneralConfig.getId().getConfigGroup());
+                    generalConfig.setKey(jpaGeneralConfig.getId().getGcKey());
+                    generalConfig.setValue(jpaGeneralConfig.getGcValue());
+                    generalConfig.setDescription(jpaGeneralConfig.getDescription());
+                    generalConfig.setDefaultVal(jpaGeneralConfig.getDefaultVal());
+                    generalConfig.setEnabled(jpaGeneralConfig.getEnabled());
+                    generalConfig.setType(jpaGeneralConfig.getType());
+
+                    generalConfigList.add(generalConfig);
+                }
+
+            }
+            session.getTransaction().commit();
+        } catch (MetadataException e) {
+            session.getTransaction().rollback();
+            LOGGER.error(e);
+        } finally {
+            session.close();
+        }
+        return generalConfigList;
+    }
+
     public com.wipro.ats.bdre.md.beans.table.GeneralConfig GetGenConfigProperty(String configGroup, String key) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00100)